modules/component/src/core/utils/get-change-detection-handling.ts

```diff
@@ -0,0 +1,22 @@
+import {
+  ChangeDetectorRef,
+  NgZone,
+  ɵdetectChanges as detectChanges,
+  ɵmarkDirty as markDirty,
+} from '@angular/core';
+
+import { isIvy } from './is-ivy';
+import { hasZone } from './has-zone';
+
+export function getChangeDetectionHandler(
+  ngZone: NgZone,
+  cdRef: ChangeDetectorRef
+): <T>(component?: T) => void {
+  if (isIvy()) {
+    return !hasZone(ngZone) ? detectChanges : markDirty;
```

Comment:
Just my two cents: all the places where hasZone is used are negated, it could be improved :+1:

```suggestion
    return hasZone(ngZone) ? markDirty : detectChanges;
```

Reply:
done

Reply:
this is only one, the rest of hasZone's call are negated too